### PR TITLE
2.4 - Update Naming, Documentation & EF Core Support

### PR DESCRIPTION
**Pull request by SakuraIsayeki**

### CODE REVIEW

File: Plugin.cs [hunk 1]
```diff
@@ -19,63 +19,63 @@ public abstract class Plugin
...
 		/// Reflects the Plugin's Assembly Name.
 		/// </summary>
-		public string PluginAssemblyName { get; private set; }
+		public string AssemblyName { get; private set; }
 
 		/// <summary>
 		/// Defines Plugin's Display Name.
 		/// </summary>
-		public abstract string PluginDisplayName { get; }
+		public abstract string DisplayName { get; }
```
Comment (VelvetToroyashi) [inline Plugin.cs:32]: This could be PluginName, but I digress

File: Tools/Data/IDatabaseProvider.cs [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+/**
+ *	IDatabaseProvider.cs
+ *	Licensed by YumeChan-DT (Nodsoft Systems) under GNU-LGPL v2.1
+ **/
+
 namespace YumeChan.PluginBase.Tools.Data
 {
-	
+	/// <summary>
+	/// Provides a database handler, for a plugin manifest of type <see cref="TPlugin"/>.
```
Comment (VelvetToroyashi) [inline Tools/Data/IDatabaseProvider.cs:12]: Database *handler* isn't very apt; it's somewhat of an abstraction.

**Review by VelvetToroyashi — APPROVED** [review]
LGTM; minor wording suggestions.

File: Plugin.cs (hunk 1 above)
Comment (SakuraIsayeki) [reply Plugin.cs:32]: Marking the distinction between internal-oriented `AssemblyName`, and public-oriented `DisplayName`. PluginName might imply both...

**Pull request merged**